[BAHIR-90] Include Apache Flink extensions to website setup

tzulitai opened the pull request:
This commit setups links for documentation and downloads for Apache
Bahir Flink extensions. The Flink docs are generated using the updated `update-docs.sh` script in #8.

It also revamps the website's project infrastructure, configuration, and navigation / layouts for documentations and the download page to be more friendly for newly supported platforms in the future.

Notes about change to the website's description in the title:
I tried adding "such as Apache Spark _and Apache Flink_" at the end, but the line became too long and looked a bit weird. Therefore, I ended up with just changing to "Apache Bahir provides extensions to multiple distributed analytic platforms." I'm generally open to discuss the changes to website content wordings in this PR!

Also, the links for Flink in the new download page are obviously invalid at the moment. They'll become valid as soon as Apache Bahir for Flink 1.0 is released, and we replace the `TBA` placeholder values for `flink_latest_release` and `flink_latest_release_date`  in `project.yml`.

ckadner commented:
@tzulitai -- thanks for your PR. This looks good to me overall. Cleaner than before :-)

@lresende, @rmetzger -- what are your thoughts?

ckadner commented on site/_config.yml:
```diff
@@ -16,7 +16,7 @@
 email: dev@bahir.apache.org
 description: > # this means to ignore newlines until "baseurl:"
-  Apache Bahir provides extensions to distributed analytic platforms such as Apache Spark.
+  Apache Bahir provides extensions to multiple distributed analytic platforms.
```
I would keep mentioning Spark and Flink ... think Google search results

"Apache Bahir provides extensions to distributed analytic platforms such as Apache Spark and Apache Flink"

ckadner commented on site/_data/project.yml:
```diff
@@ -15,15 +15,14 @@
-unix_name: bahir
-github_project_name: bahir
-description: Apache Bahir provides extensions to distributed analytic platforms such as Apache Spark.
+description: Apache Bahir provides extensions to multiple distributed analytic platforms.
```
I would keep mentioning Spark and Flink ... see above

ckadner commented on site/_includes/themes/apache-clean/_navigation.html:
```diff
@@ -5,8 +5,8 @@
-    {% if site.data.project.latest_release %}
-    <a href="{{ site.data.project.download }}" class="btn btn-primary btn-lg bigFingerButton" role="button">Download {{ site.data.project.name }} {{site.data.project.latest_release}}</a></p>
+    {% if site.data.project.download %}
+      <a href="{{ site.data.project.download }}" class="btn btn-primary btn-lg bigFingerButton" role="button">Download {{ site.data.project.name }} Latest Releases</a></p>
```
I think we should add two buttons, one for Bahir Spark and one for Bahir Flink (once available) and keep the version in the button text(s)
and the test should be for `spark_latest_release` and `flink_latest_release` (barring "TBA", of course)

ckadner commented on site/_includes/themes/apache/_navigation.html:
```diff
@@ -49,7 +49,7 @@
     <p>{{ site.data.project.description }}</p>
 
-    {% if site.data.project.latest_release %}
+    {% if site.data.project.download %}
```
I think we should add two buttons, one for Bahir Spark and one for Bahir Flink (once available) and keep the version in the button text(s)
and the test should be for `spark_latest_release` and `flink_latest_release` (barring "TBA", of course)

ckadner commented on site/_config.yml:
```diff
@@ -16,7 +16,7 @@
+  Apache Bahir provides extensions to multiple distributed analytic platforms.
 baseurl: "" # the subpath of your site, e.g. /blog
 url: "http://bahir.apache.org" # the base hostname & protocol for your site
 #twitter_username: ApacheBahir
```
on [line 32](https://github.com/apache/bahir-website/pull/9/files#diff-0a55ba953b4a00424ce1150c8cb5e4efR32), this: `exclude: [README.md,Gemfile*,docs/spark/templates]`
should now be: `exclude: [README.md,Gemfile*,docs/*/templates]` to also exclude flink templates from generation

ckadner commented on site/download.md:
```diff
@@ -69,37 +70,87 @@ Our latest release is {{ site.data.project.name }} {{site.data.project.latest_re
+
+Our latest {{ site.data.project.name }} release for Apache Flink extensions is {{site.data.project.flink_latest_release}}, released on {{site.data.project.flink_latest_release_date}}.
+
+{% if site.data.project.flink_latest_release %}
```
don't render this table if release is not published ("TBA")

ckadner commented on site/_data/project.yml:
```diff
@@ -48,10 +47,31 @@ commits_list_archive_markmail:
+flink_unix_name: bahir-flink
+flink_github_project_name: bahir-flink
+
+flink_latest_release: TBA
```
leave blank (null) to not render dependent content, like release notes, download table, download big finger button, etc

tzulitai commented:
Thanks for the comments @ckadner. They make sense! I'll update the PR accordingly.

tzulitai commented:
@ckadner -- I've hopefully addressed your comments. Please let me know if there's anything else you think we should change!

Note: Instead of allowing `TBA`, I decided to just keep it simple by checking if the `xx_latest_release` config exists.

lresende commented:
Overall LGTM, just the caveat that I haven't actually tested in a staging site.

rmetzger commented:
Thank you for merging.
